etc/kayobe/kolla/globals.yml

```
@@ -10,16 +10,20 @@ enable_docker_repo: "{% raw %}{{ 'overcloud' not in group_names or ansible_facts
 # This is necessary for os migrations where mixed clouds might be deployed
 kolla_base_distro: "{% raw %}{{ ansible_facts.distribution | lower }}{% endraw %}"
 
-kayobe_image_tags:
-  openstack:
-    rocky: zed-rocky-9-20230821T155947
-    ubuntu: zed-ubuntu-jammy-20230821T155947
-  ovn:
-    rocky: zed-rocky-9-20230925T132313
-    ubuntu: zed-ubuntu-jammy-20230821T155947
-
-openstack_tag: "{% raw %}{{ kayobe_image_tags['openstack'][kolla_base_distro] }}{% endraw %}"
-ovn_tag: "{% raw %}{{ kayobe_image_tags['ovn'][kolla_base_distro] }}{% endraw %}"
+# Dict of Kolla image tags to deploy for each service.
+# Each key is the tag variable prefix name, and the value is another dict,
+# where the key is the OS distro and the value is the tag to deploy.
+# NOTE: This is defined in etc/kayobe/kolla-image-tags.yml.
+kolla_image_tags:
+{{ kolla_image_tags | to_nice_yaml | indent(width=4, first=true) }}
+
+{% for tag in kolla_image_tags %}
```

```suggestion
{% for tag in kolla_image_tags %}
```

Couldn't we hardcode this list. That way we could get proper precedence:

nova_libvirt_tag:  kolla_image_tags['nova-libvirt'] | default(nova_tag) | default(openstack_tag)

?

---

I guess this isn't actually an issue if you had the full list of containers :+1:

---

That's an option. I was looking at writing a script that would generate the necessary variables based on kolla_image_tags, but maybe that's trying to be too "clever"?

I'd probably want to put the variables in a separate file and copy them to "declutter" it a bit though.

---

This version is still using the groups such as nova. The conflict resolution issues in the full image list prevented it from getting finished so I thought I'd try this hybrid approach.